### PR TITLE
New dhcp vendor class fingerprints

### PR DESCRIPTION
**Pull request by modoyle-r7**
New fingerprints based on captured vi_vendor_class strings
  Dell Printers
  Mitel IP Phone
  BrightSign Media Server
  Digium D Series IP Phone
  Dell iDRAC
  Linux with Busybox service

Busybox versions: https://busybox.net/downloads/binaries/
Digium Firmware: http://downloads.digium.com/pub/telephony/res_digium_phone/firmware/

### CODE REVIEW

File: xml/dhcp_vendor_class.xml [hunk 1]
```diff
@@ -305,4 +354,38 @@
...
+    <example hw.model="XD1034-W">BrightSign XD1034-W</example>
+    <example hw.model="LS424-W">BrightSign LS424-W</example>
+    <example hw.model="XT244">BrightSign XT244</example>
+    <example hw.model="HD1023">BrightSign HD1023</example>
+    <example hw.model="XT1144-PP">BrightSign XT1144-PP</example>
+    <param pos="0" name="hw.vendor" value="BrightSign"/>
+    <param pos="0" name="hw.device" value="Media Server"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="os.vendor" value="BrightSign"/>
+    <param pos="0" name="os.family" value="Linux"/>
```
Comment (mkienow-r7) [inline xml/dhcp_vendor_class.xml:370]: Based on the BrightSign CPE that exists (`o:brightsign:4k242_firmware`) this should probably have an `os.product` param.
```
<param pos="0" name="os.product" value="{hw.model} Firmware"/>
```

**Review by mkienow-r7 — CHANGES_REQUESTED** [review]

File: xml/dhcp_vendor_class.xml (hunk 1 above)
Comment (modoyle-r7) [reply xml/dhcp_vendor_class.xml:370]: Will do.

File: xml/dhcp_vendor_class.xml [hunk 2]
```diff
@@ -95,6 +95,37 @@
...
+  <fingerprint pattern="^Mfg=DELL;Typ=Printer;Mod=Dell ([A-Z]?\d{4}dn)(?:\sLaser Printer)?;Ser=([A-Z0-9]{7});$">
+    <description>Dell Laser Printer</description>
+    <example hw.model="B2360dn" hw.serial_number="ABCDEF1">Mfg=DELL;Typ=Printer;Mod=Dell B2360dn Laser Printer;Ser=ABCDEF1;</example>
+    <example hw.model="2330dn" hw.serial_number="123ABCD">Mfg=DELL;Typ=Printer;Mod=Dell 2330dn Laser Printer;Ser=123ABCD;</example>
+    <example hw.model="S2830dn" hw.serial_number="1A2BC34">Mfg=DELL;Typ=Printer;Mod=Dell S2830dn;Ser=1A2BC34;</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Dell"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="2" name="hw.serial_number"/>
+    <param pos="0" name="os.vendor" value="Dell"/>
```
[the remark this answers is not in the document]
Comment (modoyle-r7) [reply xml/dhcp_vendor_class.xml:107]: Will do.

File: xml/dhcp_vendor_class.xml [hunk 3]
```diff
@@ -305,4 +359,40 @@
...
+    <description>BrightSign Digital Signage Player</description>
+    <example hw.model="HS124">BrightSign HS124</example>
+    <example hw.model="XD1034">BrightSign XD1034</example>
+    <example hw.model="XD1034-W">BrightSign XD1034-W</example>
+    <example hw.model="LS424-W">BrightSign LS424-W</example>
+    <example hw.model="XT244">BrightSign XT244</example>
+    <example hw.model="HD1023">BrightSign HD1023</example>
+    <example hw.model="XT1144-PP">BrightSign XT1144-PP</example>
+    <param pos="0" name="hw.vendor" value="BrightSign"/>
+    <param pos="0" name="hw.device" value="Media Server"/>
```
Comment (mkienow-r7) [inline xml/dhcp_vendor_class.xml:372]: Would "Media Player" be a more appropriate value for `*.device` than "Media Server"?
> BrightSign LLC, the global market leader in digital signage media players ...

Comment (modoyle-r7) [reply xml/dhcp_vendor_class.xml:372]: I went with Media Sever as both favicons.xml and html_title.xml databases have existing entries for BrightSign that use Media Server for hw.device.
(while also referring to the device as a Media Player in description and hw.product ?)

Comment (mkienow-r7) [reply xml/dhcp_vendor_class.xml:372]: Thank you for pointing those out as I overlooked them. Then we'll accept this as-is and change them all in a separate PR if appropriate.

**Review by mkienow-r7 — APPROVED** [review]

**Comment by mkienow-r7** [comment]
Thank you for the contribution @modoyle-r7!

**Pull request merged**